utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,83 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
 }
 
 impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-    /// Generates a function which calls "properly" the `gr_reservation_send`.
-    fn generate_send_from_reservation(&mut self) {
-        let (reserve_gas_idx, reservation_send_idx) = {
-            let maybe_reserve_gas = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReserveGas))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-            let maybe_reservation_send = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReservationSend))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
-                None => {
-                    log::trace!("Wasm hasn't got either gr_reserve_gas, or gr_reservation_send, or both of them");
-                    return;
-                }
-            }
-        };
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes(&self, sys_calls: &[SysCallName]) -> Option<Vec<usize>> {
+        let ret: Option<Vec<_>> = sys_calls
+            .iter()
+            .map(|&sys_call| {
+                self.sys_calls_imports
+                    .get(&InvocableSysCall::Loose(sys_call))
+                    .map(|&(_, call_indexes_handle)| call_indexes_handle)
+            })
+            .collect();
+
+        if ret.is_none() {
+            log::trace!(
+                "Wasm should import the following syscalls: {missing_sys_calls}",
```

Comment:
Also log which "precise" sys-call invocation required that. Currently there's only `gr_reservation_send`.

Reply:
This would require defining the precise sys-call at the top level and storing it in a local variable and then passing it to this function. It doesn't look good, but I can do it

Reply:
done in faf88bd231a295964785a02d4d68a1a894e98104